lib/websocket/socket-base-impl.ts

```diff
@@ -0,0 +1,89 @@
+/// <reference path='../../typings/tsd.d.ts' />
+
+import bunyan = require('bunyan');
+import blpapi = require('../blpapi-wrapper');
+import Interface = require('../interface');
+import SocketEventEmitterAdapter = require('./socket-event-emitter-adapter');
+
+export = SocketBaseImpl;
+
+class SocketBaseImpl extends SocketEventEmitterAdapter implements Interface.ISocket {
+
+    // DATA
+    private _logger: bunyan.Logger;
+    private _blpapiSession: blpapi.Session;
+
+    // PROTECTED MANIPULATORS
+    protected send(name: string, ...args: any[]): void {
+        throw new Error('send() must be implemented');
+    }
+
+    // CREATORS
+    constructor(logger: bunyan.Logger, emitterToAdapt?: SocketIO.Socket) {
```

Comment:
I prefer using default parameters for simplicity, so we can remove all the optional parameters and argument length check.
```typescript
constructor(logger: bunyan.Logger, emitterToAdapt: SocketIO.Socket | events.EventEmitter = new events.EventEmitter()) {
```